core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -1502,10 +1502,12 @@ private static void findCorrelationEquivalent(CorRef correlation, RexNode e)
     case EQUALS:
       final RexCall call = (RexCall) e;
       final List<RexNode> operands = call.getOperands();
-      if (references(operands.get(0), correlation)) {
+      if (!RexUtil.containsFieldAccess(operands.get(1))
```

Comment:
I think that the JavaDoc of this function needs to be changed; it no longer reflects what it is doing.